docs/taskcluster-yml-v1.md

```diff
@@ -0,0 +1,189 @@
+  * `pull_request.labeled`
+  * `pull_request.unlabeled`
+  * `pull_request.opened`
+  * `pull_request.edited`
+  * `pull_request.closed`
+  * `pull_request.reopened`
+  * `pull_request.synchronize` (a new commit is pushed to the branch in the PR. NOTE: There is no 'd' at the end of 'synchronize')
+  * `push`                     (a push is made directly to the repo)
+  * `release`                  (a new release published in any branch of the repo.)
+  * `tag`                      (a tag is pushed to the repo)
+
+In almost all cases, you'll only want `[push, pull_request.opened, pull_request.synchronize]`.
+
+### Branch Filtering
+
+You can also modify a task definition so that it will only run for events on certain branches. For example, the task defined below will only run for pushes to the master branch:
+
+```yaml
+version: 1
+tasks:
+  - ...
+    extra:
+      github:
+        events:
+          - push
+        branches:
+          - master
+```
+
+You can even choose to exclude certain branches from your task run. The task will run only if it is not on one of the excluded branches. For example, the task defined below will only run for pushes **not** to the master branch:
+
+```yaml
+version: 1
+tasks:
+  - ...
+    extra:
+      github:
+        events:
+          - push
+        excludeBranches:
+          - master
+```
+
+You may not specify both an `excludeBranches` and `branches` at the same time.
+
+# Scopes and Roles
+
+[Roles](https://docs.taskcluster.net/manual/design/apis/hawk/roles) are, in a nutshell, sets of [scopes](https://docs.taskcluster.net/reference/platform/taskcluster-auth/docs/scopes). Taskcluster-Github uses a very specific role to create tasks for each project.  That role has the form
+* `assume:repo:github.com/<owner>/<repo>:branch:<branch>` for a push event
+* `assume:repo:github.com/<owner>/<repo>:pull-request` for a pull request
+* `assume:repo:github.com/<owner>/<repo>:release` for a release event
+
+In the [role manager](https://tools.taskcluster.net/auth/roles/), you can set up roles however you like. To give permissions to every event in your repository, you can make a role `repo:github.com/<organization>/<repository>:*` or you can give fine-grained permissions to specific github events or specific branches.
+
+Careful configuration of these roles and the related tasks can allow powerful behaviors such as binary uploads on push, without allowing pull requests access to those capabilities. There are lots of examples in the role manager for other repositories that have been set up. Look for roles that begin with `repo:github.com/` to see how they work.
+
+# Example
+
+```yaml
+version: 1
+policy:
+  pullRequests: public
+metadata:
+  name: "TaskCluster GitHub Tests"
+  description: "All non-integration tests for taskcluster github"
+  owner: ${event.head.user.email}
+  source: ${event.head.repo.url}
+tasks:
+  - provisionerId: ${taskcluster.docker.provisionerId}
+    workerType: ${taskcluster.docker.workerType}
```

Comment:
I think we want to get away from supplying these values.  We have never changed them, and probably never will, so we might as well just tell our users what provisionerId / workerType to use.

Reply:
Totally agreed.

Reply:
Sounds cool!